include/sys/thread_stack.h

```diff
@@ -50,8 +50,45 @@ struct __packed z_thread_stack_element {
  * Take the provided value and round it down such that the value is aligned
  * to the CPU and ABI requirements. This is not used for any memory protection
  * hardware requirements.
+ *
+ * @param ptr Proposed stack pointer address
+ * @return Properly aligned stack pointer address
+ */
+static inline char *z_stack_ptr_align(char *ptr)
+{
+	return (char *)ROUND_DOWN(ptr, ARCH_STACK_PTR_ALIGN);
+}
+#define Z_STACK_PTR_ALIGN(ptr) ((uintptr_t)z_stack_ptr_align((char *)(ptr)))
```

Comment:
MISRA-C would probably happier if we use `u8_t*` or `uint8_t*` since `char` is signed. This is not a blocker just commenting it here to see if there is  anything against using `u8_t` instead of `char` in these situations.

Reply:
sure I can change to u8_t *

Reply:
Hi @ceolin I'd like to punt on this part, as there is a bunch of other existing stack code currently using char * as the data type for the raw stack buffer. Can you file a GH issue on this and I'll get to it in another PR?

Reply:
yep, I can track it in a MISRA-C issue. This is not the only place doing it.